src/@aml/@aml_create.jl

```diff
@@ -73,6 +73,11 @@ function aml_create(expr::Expr, args_param, args_defaultvalue, args_type, args_v
 
         end # endfor
         ################################################################
+        # Remove supertype if it is present
+        if T isa Expr && T.head == :<:
+            T = T.args[1]
```

Comment:
This is fine but probably should give a warning because we are not doing anything with the supertype here. 

I think I remember I wanted to add proper support for supertypes, but it was a breaking feature (if I recall correctly). Maybe I should see if we can add them properly (rather than removing)

Reply:
I think what happens is the supertype passes thru (via `expr`) so it still participates in the final `struct` definition, but is ignored for the AcuteML machinery. I don't check for this in the tests, but should. Is this the behavior you'd expect?

Reply:
see 3760114 for the test I missed

Reply:
Thanks for the tests. Yeah, it seems to work, but could you try to move the code to line 99. Just for consistency with other conditions.
```julia
        elseif T isa Expr && T.head == :<:
```
https://github.com/alhirzel/AcuteML.jl/blob/376011465ccebb80195ef600e6e3713a5be807da/src/%40aml/%40aml_create.jl#L99

Reply:
or this needs to be kept in a separate condition?

Reply:
Oh. OK. It needs to be seprate. I am remembering what is happening here.